src/index.js

```diff
@@ -1,10 +1,167 @@
 import { months } from './data.js';
 
 (async function () {
-  const allCityDetails = await fetchCityDetails()
+  let allCityDetails = await fetchCityDetails()
+  const icons = document.querySelectorAll('.mid-ic')
+  const leftArrow = document.getElementById('left-scroll-arrow')
+  const rightArrow = document.getElementById('right-scroll-arrow')
+  let middleSegmentCards = allCityDetails
+  let selectedIcon = ' '
+  let cardCount = document.getElementById('select-icon').value
+  allCityDetails = sortCityOptions(allCityDetails)
   loadSelectOptions(allCityDetails)
   changeHeaderValues()
   document.querySelector('.option-select.city').addEventListener('change', changeHeaderValues)
+  changeMiddleSegment(allCityDetails)
+  getScrollWidth()
+  icons.forEach((ic) => ic.addEventListener('click', selectIcon))
```

Comment:
Not sure what this icons mean, would be good like weatherList or something similar